workflow/workflow_test.go

```diff
@@ -410,6 +418,247 @@ func TestWaitForStarted(t *testing.T) {
 	assert.Equals(t, outputID, "success")
 }
 
+var waitForSerialForeachWf = `
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject 
+      properties: {}
+steps:
+  second_wait:
+    wait_for: !expr $.steps.first_wait.outputs.success
+    kind: foreach
+    parallelism: 3
+    items: 
+    - wait_time_ms: 10
+    - wait_time_ms: 10
+    - wait_time_ms: 10
```

Comment:
Why are you using a parallelism 3 sub-workflow? If it doesn't work as intended, it could result in more execution time that delays it, making the test less effective.